orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -118,7 +118,7 @@ def _forward(self, *inputs, **kwargs):
                 self._input_names_require_grad = input_names_require_grad
                 self._initialize_module_gradient_graph_builder()
 
-            if self._current_input_shape is None or self._current_input_shape != new_input_shape:
+            if self._current_input_shape is None:
                 self._current_input_shape = new_input_shape
```

Comment:
`self._current_input_shape` will become obsolete and lead to confusion. If we don't care about the exact shape, this should be a boolean